src/modules/Dropdown/Dropdown.js

```diff
@@ -687,6 +693,8 @@ export default class Dropdown extends Component {
   // ----------------------------------------
 
   renderText = () => {
+    if (this.props.trigger) return this.props.trigger
```

Comment:
Ah, you're totally right.  Elsewhere, we've been using early conditional returns to only render what is necessary.  It also strictly prevents people from incorrectly using the component since it won't render incorrect usages.  What about this?

```jsx
    if (trigger) {
      return (
        <ElementType
          {...rest}
          {...conditionalProps}
          onChange={this.onChange}
          onClick={this.handleClick}
          className={classes}
        >
          {trigger}
          {createIcon(icon)}
          {this.renderMenu()}
        </ElementType>
      )
    }

    return (
      <ElementType
        {...rest}
        {...conditionalProps}
        onChange={this.onChange}
        onClick={this.handleClick}
        className={classes}
      >
        {this.renderHiddenInput()}
        {this.renderLabels()}
        {this.renderSearchInput()}
        {this.renderSearchSizer()}
        {this.renderText()}
        {createIcon(icon)}
        {this.renderMenu()}
      </ElementType>
    )
```